src/legacy-routes.js

```diff
@@ -13,10 +13,19 @@ const invalidTx = 'Invalid transaction id!'
 
 const arraySum = (numbers) => numbers.reduce((acc, val) => acc.plus(Big(val)), Big(0))
 
-const txAddressCoins = (addresses, amounts, address) => arraySum(zip(addresses, amounts)
-  .filter((pair) => pair[0] === address)
+const txAddressCoins = (addresses, amounts, addressSet) => arraySum(zip(addresses, amounts)
```

Comment:
I would add there a comment explaining that this function takes movements for various addresses and returns their sum filtered by a certain set of addresses we are interested in.

Took me a while to understand why addresses are being passed in two arguments and I don't see a straightforward way to refactor this.

Reply:
I added an explanatory comment.